src/asynqp/connection.py

```diff
@@ -60,9 +60,10 @@ def close(self):
 
         This method is a :ref:`coroutine <coroutine>`.
         """
-        self.sender.send_Close(0, 'Connection closed by application', 0, 0)
-        yield from self.synchroniser.await(spec.ConnectionCloseOK)
-        self.closed.set_result(True)
+        if not self.closed.done():
+            self.sender.send_Close(0, 'Connection closed by application', 0, 0)
+            yield from self.synchroniser.await(spec.ConnectionCloseOK)
+            self.closed.set_result(True)
```

Comment:
Don't see any valid use-case where you would like to be not able to call close 2-ce. It does make sense to add a ``is_closed()`` method thou.
Current behaviour of locking and raising an exception is bad, as the connection can be closed gracefully server side in AMQP protocol. If it will raise an exception, you will **always** need to wrap the call in ``try/except`` as you will never know if the server did not close the connection on it's own.